src/nvim/diff.c

```diff
@@ -2627,26 +2622,66 @@ void ex_diffgetput(exarg_T *eap)
     }
   }
 
+  diffgetput(eap->addr_count, idx_cur, idx_from, idx_to, eap->line1, eap->line2);
+
+  // restore curwin/curbuf and a few other things
+  if (eap->cmdidx != CMD_diffget) {
+    // Syncing undo only works for the current buffer, but we change
+    // another buffer.  Sync undo if the command was typed.  This isn't
+    // 100% right when ":diffput" is used in a function or mapping.
+    if (KeyTyped) {
+      u_sync(false);
+    }
+    aucmd_restbuf(&aco);
+  }
+
+theend:
+  diff_busy = false;
+
+  if (diff_need_update) {
+    ex_diffupdate(NULL);
+  }
+
+  // Check that the cursor is on a valid character and update its
+  // position.  When there were filler lines the topline has become
+  // invalid.
+  check_cursor();
+  changed_line_abv_curs();
+
+  if (diff_need_update) {
+    // redraw already done by ex_diffupdate()
+    diff_need_update = false;
+  } else {
+    // Also need to redraw the other buffers.
+    diff_redraw(false);
+    apply_autocmds(EVENT_DIFFUPDATED, NULL, NULL, false, curbuf);
+  }
+}
+
+static void diffgetput(const int addr_count, const int idx_cur, const int idx_from,
```

Comment:
Add a comment about what this function does?

Reply:
Actually, naming this function `diffgetput()` is misleading, as it only does the "get". `ex_diffgetput()` is responsible for turning a "put" into a "get".